src/Microsoft.AspNet.Mvc.Core/Formatters/HttpNotAcceptableOutputFormatter.cs

```diff
@@ -1,40 +0,0 @@
-// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
-// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
-
-using System;
-using System.Collections.Generic;
-using System.Threading.Tasks;
-using Microsoft.AspNet.WebUtilities;
-using Microsoft.Net.Http.Headers;
-
-namespace Microsoft.AspNet.Mvc
-{
-    /// <summary>
-    /// A formatter which does not have a supported content type and selects itself if no content type is passed to it.
-    /// Sets the status code to 406 if selected.
-    /// </summary>
-    public class HttpNotAcceptableOutputFormatter : IOutputFormatter
```

Comment:
Let's discuss this more tomorrow. `MatchOnObjectType` seems like a hammer that you can easily hurt yourself with.